chain/chunks/src/client.rs

```diff
@@ -115,57 +114,12 @@ impl ShardedTransactionPool {
         }
         reintroduced_count
     }
-
-    /// Migrate all of the transactions in the pool from the old shard layout to
-    /// the new shard layout.
-    /// It works by emptying the pools for old shard uids and re-inserting the
-    /// transactions back to the pool with the new shard uids.
-    pub fn reshard(&mut self, old_shard_layout: &ShardLayout, new_shard_layout: &ShardLayout) {
```

Comment:
It feels like we will need transaction pool resharding for IR too.

Reply:
Yeah, thought so, maybe with some slightly changes

Let me know if I should keep this function alive

Reply:
Yeah, I think it's easier to keep it for now.

Reply:
Done!